pkg/mountutil/mountutil.go

```diff
@@ -49,90 +53,60 @@ type Processed struct {
-			res.Name = src
-			vol, err := volStore.Get(src, false)
-			if err != nil {
-				if errors.Is(err, errdefs.ErrNotFound) {
-					vol, err = volStore.Create(src, nil)
-					if err != nil {
-						return nil, fmt.Errorf("failed to create volume %q: %w", src, err)
-					}
-				} else {
-					return nil, fmt.Errorf("failed to get volume %q: %w", src, err)
-				}
-			}
-			// src is now full path
-			src = vol.Mountpoint
-			res.Type = Volume
-		}
-		if !filepath.IsAbs(src) {
-			log.L.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", src)
-			var err error
-			src, err = filepath.Abs(src)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get the absolute path of %q: %w", src, err)
-			}
-		}
-		if createDir {
-			if _, err := os.Stat(src); err != nil {
-				if !os.IsNotExist(err) {
-					return nil, fmt.Errorf("failed to stat %q: %w", src, err)
-				}
-				if err := os.MkdirAll(src, 0o755); err != nil {
-					return nil, fmt.Errorf("failed to mkdir %q: %w", src, err)
-				}
-			}
-		}
+type volumeSpec struct {
+	Type            string
+	Name            string
+	Source          string
+	AnonymousVolume string
+}
 
-		if !filepath.IsAbs(dst) {
-			return nil, fmt.Errorf("expected an absolute path, got %q", dst)
-		}
-		rawOpts := ""
-		if len(split) == 3 {
-			rawOpts = split[2]
-		}
-		res.Mode = rawOpts
+func ProcessFlagV(s string, volStore volumestore.VolumeStore, createDir bool) (*Processed, error) {
+	split, err := splitVolumeSpec(s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to split volume mount specification %q: %v", s, err)
+	}
 
-		// always call parseVolumeOptions for bind mount to allow the parser to add some default options
-		var err error
-		var specOpts []oci.SpecOpts
-		options, specOpts, err = parseVolumeOptions(res.Type, src, rawOpts)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse volume options (%q, %q, %q): %w", res.Type, src, rawOpts, err)
-		}
-		res.Opts = append(res.Opts, specOpts...)
-	default:
+	if len(split) < 1 || len(split) > 3 {
 		return nil, fmt.Errorf("failed to parse %q", s)
 	}
 
-	fstype := "nullfs"
+	src := split[0]
+	dst, err := getDestination(split)
+	if err != nil {
+		return nil, err
+	}
+
+	isAnonymousVol := len(split) == 1
+	volSpec, err := handleVolumeToMount(src, volStore, createDir, isAnonymousVol)
+	if err != nil {
+		return nil, err
+	}
+
+	src = volSpec.Source
+	res := &Processed{
+		Type:            volSpec.Type,
+		Name:            volSpec.Name,
+		AnonymousVolume: volSpec.AnonymousVolume,
+	}
+
+	// This is to ensure that the destination path is an absolute path
+	if isAnonymousVol {
+		dst = src
+	}
+
+	if len(split) == 3 {
+		res.Mode = split[2]
+	}
+
+	rawOpts := res.Mode
+	options, spectOpts, err := getVolumeOptions(src, res.Type, rawOpts)
+	if err != nil {
+		return nil, err
+	}
+	res.Opts = spectOpts
+
+	fstype := DefaultMountType
 	if runtime.GOOS != "freebsd" {
```

Comment:
We should ideally strive to have ZERO platform-dependant code in the main `mountutil.go` file, so could you please move this logic into a  platform-specific function like `ensureBindOption()`?

Reply:
While striving for ZERO platform-dependant code is ideal, in some cases it may not apply. The code works for both Windows and Linux and creating a file for non-freebsd code with a single function does not add any value. Should there be future requirements for a non-freebsd platform, then we could create the file then